### PR TITLE
Fix crash in CRD display

### LINKED ISSUE #3100: App crash at http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/crd/datadoghq.com/watermarkpodautoscalers
View carshes on viewing crd/datadoghq.com/watermarkpodautoscalers
Doing the same with kubectl get works

```
 http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7552631
    at div
    at div
    at t (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7804646)
    at div
    at qh (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7812615)
    at div
    at ip (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7814798)
    at div
    at div
    at ny (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7964629)
    at Ay (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7973947)
    at UE (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7935255)
    at t (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:227449)
    at t (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:229529)
    at cc (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7682615)
    at main
    at div
    at http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7552631
    at Lk (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:8266983)
    at t (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:227449)
    at t (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:229529)
    at cc (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7682615)
    at main
    at div
    at pU (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:8513301)
    at cc (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7682615)
    at t (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:223994)
    at BU (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:8514805)
```

Error stack:

 
```
Error: Parse error on line 1:
...ernal.currentValue..
-----------------------^
Expecting 'STAR', 'IDENTIFIER', 'SCRIPT_EXPRESSION', 'INTEGER', 'END', '[', got '1'
    at i.parseError (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:783658)
    at i.e.yy.parseError (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:804223)
    at i.parse (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:784753)
    at o._normalize (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:803401)
    at o.stringify (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:803009)
    at o.value (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:801590)
    at http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7936416
    at Array.map (<anonymous>)
    at renderTableContents (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7936397)
    at ny.getRow (http://cc6a06a0-a3e8-4fec-b8a5-c1ba80be635e.localhost:63423/build/Lens.js:2:7966830)
```

### LINKED ISSUE #4999: Crash by Parse.parseError
Component stack:

    at CrdResourceDetails (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:246735:47)
    at div
    at div
    at div
    at Animate (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:148420:9)
    at drawer_Drawer (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:14197:9)
    at KubeObjectDetails (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:289362:9)
    at ErrorBoundary (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:238009:9)
    at Router (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:79008:30)
    at ClusterFrame (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/chunks/5.js:7940:9)
    at ThemeProvider (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:609896:24)
Error stack:

Error: Parse error on line 1:
...ernal.currentValue..
-----------------------^
Expecting 'STAR', 'IDENTIFIER', 'SCRIPT_EXPRESSION', 'INTEGER', 'END', '[', got '1'
    at Parser.parseError (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:227349:15)
    at Parser.parser.yy.parseError (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:228549:17)
    at Parser.parse (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:227407:22)
    at JSONPath._normalize (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:228483:24)
    at JSONPath.stringify (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:228453:15)
    at JSONPath.value (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:228348:31)
    at http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:246737:201
    at Array.map (<anonymous>)
    at CrdResourceDetails.renderAdditionalColumns (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:246737:24)
    at CrdResourceDetails.render (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:246767:18)
    at allowStateChanges (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:2571:12)
    at http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:10958:59
    at trackDerivedFunction (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:3120:18)
    at Reaction.track (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:3649:18)
    at CrdResourceDetails.reactiveRender [as render] (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:10956:14)
    at CrdResourceDetails.makeComponentReactive (http://e1f1df19d99c982b330d4c7a6ba292bb.localhost:59974/build/Lens.js:10971:25)

### LINKED ISSUE #5252: ApisixRoute Custom Resource Can Not be Loaded
**Describe the bug**
ApisixRoute Custom Resource can not be loaded.

**To Reproduce**
1. Install Apache APISIX Ingress
2. Check ApisixRoute resource in Lens Custom Resources

**Expected behavior**
The CR should be listed.

**Screenshots**
image

**Environment (please complete the following information):**
- Lens Version: 5.4.5-latest

**Logs:**
When click the CR menu, Lens results in error.
```
Error stack:

Error: Parse error on line 1:
spec.http[].match.hosts
----------^
Expecting 'STAR', 'SCRIPT_EXPRESSION', 'INTEGER', 'ARRAY_SLICE', 'FILTER_EXPRESSION', 'QQ_STRING', 'Q_STRING', got ']'
    at Parser.parseError (http://f247efbdaa8840b88852817bab182b01.localhost:61370/build/Lens.js:223941:15)
    at Parser.parser.yy.parseError (http://f247efbdaa8840b88852817bab182b01.localhost:61370/build/Lens.js:225141:17)
    at Parser.parse (http://f247efbdaa8840b88852817bab182b01.localhost:61370/build/Lens.js:223999:22)
    at JSONPath._normalize (http://f247efbdaa8840b88852817bab182b01.localhost:61370/build/Lens.js:225075:24)
    at JSONPath.stringify (http://f247efbdaa8840b88852817bab182b01.localhost:61370/build/Lens.js:225045:15)
    at JSONPath.value (http://f247efbdaa8840b88852817bab182b01.localhost:61370/build/Lens.js:224940:31)
    at http://f247efbdaa8840b88852817bab182b01.localhost:61370/build/Lens.js:236866:52
    at Array.map (<anonymous>)
    at renderTableContents (http://f247efbdaa8840b88852817bab182b01.localhost:61370/build/Lens.js:236865:33)
    at ItemListLayoutContent.getTableRow (http://f247efbdaa8840b88852817bab182b01.localhost:61370/build/Lens.js:137069:13)
```

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes https://github.com/lensapp/lens/issues/4999
fixes https://github.com/lensapp/lens/issues/5252
fixes https://github.com/lensapp/lens/issues/3100

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -199,6 +199,7 @@
     }
   },
   "dependencies": {
+    "@astronautlabs/jsonpath": "^1.1.0",
```
Comment (ixrock) [inline package.json:202]: What the differences with https://www.npmjs.com/package/jsonpath ? _(it has more weekly downloads)_

File: src/renderer/utils/jsonPath.ts [hunk 2]
```diff
@@ -38,3 +73,35 @@ function convertToIndexNotation(key: string, firstItem = false) {
     return `${prefix}${key}`;
   }
 }
+
+function formatJSONValue(value: unknown) {
```
Comment (ixrock) [inline src/renderer/utils/jsonPath.ts:77]: So, format should just remove quotes from a `string` value or `Date` values also supported?

btw: `JSON.stringify(value)` works the same as `String(value)` for non-object values except strings. 
but for `Date` instances its not equal:

image

**Review by ixrock — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (Nokel81) [reply package.json:202]: That package is unmaintained. https://github.com/dchester/jsonpath/issues/139

File: src/renderer/utils/jsonPath.ts (hunk 2 above)
Comment (Nokel81) [reply src/renderer/utils/jsonPath.ts:77]: Fair enough, the previous situation was to always use `JSON.stringify` though. Also in the current use case of this where we are inspecting custom resource instances I do not believe that we will ever actually encounter `Date` instances.

File: src/renderer/utils/__tests__/jsonPath.test.tsx [hunk 3]
```diff
@@ -3,44 +3,166 @@
...
+    expect(res).toBe("$.metadata.labels['serving.knative.dev/configuration']");
+  });
+
+  it("converts all [] to [0]", () => {
+    const res = convertKubectlJsonPathToNodeJsonPath(".metadata.labels[].foo[]");
+
+    expect(res).toBe("$.metadata.labels[0].foo[0]");
+  });
+
+  it("converts ending .. to ..*", () => {
```
Comment (jim-docker) [inline src/renderer/utils/__tests__/jsonPath.test.tsx:57]: Is this the right description?

**Review by jim-docker — APPROVED** [review]
verified it fixes #5252

File: src/renderer/utils/__tests__/jsonPath.test.tsx (hunk 3 above)
Comment (Nokel81) [reply src/renderer/utils/__tests__/jsonPath.test.tsx:57]: No good catch. Will fix in a follow up PR

**Pull request merged**